### ISSUE TITLE
[Question] Is it possible that the rate gets calculated automatically on import?

### ISSUE THREAD

**Issue by codiflow** [kevinpapst/ImportBundle#13]
While I'm preparing the migration of my old time sheets to Kimai I came across the fact that the rate **is not being calculated automatically** during import.

So my current workflow looks like this:
- I have to turn my old time entries with a regex into a CSV template
- I open this CSV in LibreOffice Calc (German language settings in LibreOffice) and add two formulars for calculating the `duration` in seconds from the `From` and `To` fields (`=RUNDEN(ZAHLWERT(C2-B2)*86400)`), taking the value from this field and the `hourlyrate` field to calculate the `rate` field (`=D2/60/60*M2`). As this value needs to be in the US format (`.` instead of `,`) I have to change the display format accordingly too
- After all those changes the imported entries look exactly like a new entry I would have created in Kimai

If I don't specify the field `rate`, it is set to 0 in Kimai. The same goes for the `hourlyrate` field. My expectation (and also my hope) was, that those values would be picked from the customer / project / activity automatically like how Kimai behaves when I create a new time entry from the GUI.

If I edit the imported entry (where the `rate` field was set to 0 automatically because it was not specified) and change something the rate field gets (re)calculated automatically. But that's not a solution for me because I don't want to touch each entry manually.

So the **question** is:
Is there a way to prevent the necessity to precalculate the rate before importing the timesheet into Kimai and letting Kimai calculate this field automatically like it does if I create a new entry using the GUI?

**Comment by kevinpapst** [comment]
Doesn't really make sense for me. Prices change over time. Rates should be available for your old entries.

Having more options is always nice, but the importer is one of the most used bundles and nobody ever asked for that feature. 

You can try to deactivate that line:
https://github.com/kevinpapst/ImportBundle/blob/main/Importer/ImporterService.php#L63

**Comment by codiflow** [comment]
Hi Kevin,

yes the rates are available as I described. And as they are specified on customer side they are also easily changeable for me in the future if thats necessary some day.

But for the import I have to calculate the rate depending on the amount of hours and the given HourlyRate value. My assumption was that if I specify the HourlyRate and leave the rate field blank the Importer calculates the value itself.

But it's fine for me if it doesn't – just wanted to make sure that I did not miss something here as I couldn't find anything about this behaviour in the documentation 😊

What exactly do I achieve by commenting out the line you mentioned? Is this line deactivating the Listener for the "auto complete function"? I was reading the source code twice but it's not really clear to me what this DataSubscriberInterface exactly does. A short hint would be great, thanks 😎
